src/core/constraint_template.jl

```diff
@@ -201,3 +219,81 @@ function constraint_tp_storage_exchange(pm::GenericPowerModel, i::Int; nw::Int=p
     PMs.constraint_storage_complementarity(pm, nw, i)
     constraint_tp_storage_loss(pm, nw, i, storage["storage_bus"], storage["r"], storage["x"], storage["standby_loss"])
 end
+
+
+"Links the voltage at both windings of a fixed tap transformer."
+function constraint_tp_trans_voltage(pm::GenericPowerModel, i::Int; nw::Int=pm.cnw)
+    (Tv_fr,Tv_im,Ti_fr,Ti_im,Cv_to) = calc_tp_trans_Tvi(pm, i)
+    # TODO add checks to simplify
+    f_bus = ref(pm, :trans, i)["f_bus"]
+    t_bus = ref(pm, :trans, i)["t_bus"]
+    tapset = ref(pm, :trans, i)["tapset"]
+    constraint_tp_trans_voltage(pm, i, f_bus, t_bus, tapset, Tv_fr, Tv_im, Cv_to)
+end
+
+
+"Links the voltage at both windings of a variable tap transformer."
+function constraint_tp_trans_voltage_var(pm::GenericPowerModel, i::Int; nw::Int=pm.cnw)
+    (Tv_fr,Tv_im,Ti_fr,Ti_im,Cv_to) = calc_tp_trans_Tvi(pm, i)
+    # TODO add checks to simplify
+    trans = ref(pm, :trans, i)
+    f_bus = trans["f_bus"]
+    t_bus = trans["t_bus"]
+    constraint_tp_trans_voltage_var(pm, i, f_bus, t_bus, Tv_fr, Tv_im, Cv_to)
+    # fix the taps with a constraint which are not free
+    # at least one tap will be free
+    tapfix = trans["tapfix"]
+    tapset = trans["tapset"]
+    constraint_tp_trans_tap_fix(pm, i, tapfix, tapset)
```

Comment:
Agreed. Proposed changes:
- In `tp_pf` and `tp_opf`, all transformers are treated as non-dispatchable components (labelled as trans).
- In `tp_opf_oltc`, all transformers are treated as dispatchable components (labelled as oltc).

So if I understand correctly the behaviour of the PR in PowerModels, a transformer in the data model would be split over two components in the reference dict, oltc and oltc_const/trans? Given the above changes, this would be easy to integrate later on.